File: 1537-maximum-score-after-splitting-a-string/1537-maximum-score-after-splitting-a-string.py
class Solution:
    def maxScore(self, s: str) -> int:
        zeros = 0
        ones = 0
        res = -1
        for i in s:
            if i == '1':
                ones += 1
        
        for i in range(0,len(s) - 1):
            if s[i] == '0':
                zeros += 1
            else:
                ones -= 1
            res = max(res, ones + zeros)
        return res

File: 1537-maximum-score-after-splitting-a-string/1537-maximum-score-after-splitting-a-string.java
class Solution {
    public int maxScore(String s) {
        int zeros = 0,ones = 0,res = -1,n = s.length();
        for(int i = 0;i < n;i++){
            if(s.charAt(i) == '1')ones++;
        }
        for(int i = 0;i < n -1;i++){
            if(s.charAt(i) == '0')zeros++;
            else ones--;
            res = Math.max(res,ones+zeros);
        }
        return res;
    }
}